### PR TITLE
feat: sharabale configs

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
FEATURE

**Did you add tests for your changes?**
yes, more tests WIP
**If relevant, did you update the documentation?**
WIP
**Summary**
Fixes  https://github.com/webpack/webpack-cli/issues/2748

**Does this PR introduce a breaking change?**
No

**Other information**

WIP in covering more edge cases with multiple nested configs and multi compiler mode.

### CODE REVIEW

**Review by alexander-akait — COMMENTED** [review]
Let's add tests with `--merge` too

**Comment by snitin315** [comment]
I am also implementing `--extends` flag with this.

**Comment by alexander-akait** [comment]
Yep, it is great idea

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -1691,6 +1703,35 @@ class WebpackCLI {
...
+
+                configOptions.extends.map(async (value) => {
+                    newOptions.config.push(path.resolve(value));
+                });
+
+                delete configOptions.extends;
+
+                await this.resolveConfig(newOptions);
+            }
+        });
```
Comment (alexander-akait) [inline packages/webpack-cli/lib/webpack-cli.js:1733]: We do not use `webpack-merge` logic here? it is working good

Comment (snitin315) [reply packages/webpack-cli/lib/webpack-cli.js:1733]: We will also need to load the extended configurations, here we have only `paths` for extended configurations and not their `options`. 

Or am I missing something here?

Comment (alexander-akait) [reply packages/webpack-cli/lib/webpack-cli.js:1733]: Ideally `--extends` means `--config=webpack.config.js --config=webpack.other.config.js --merge`, so I think we already have logic for this, for the next major release we will change logic for multiple `--config` https://github.com/webpack/webpack-cli/blob/master/packages/webpack-cli/lib/webpack-cli.js#L1619, I think we will throw an error on two and more config options without `--merge`, but we can reuse this logic for `--extends`, `--extends` like syntax sugar, also do not forget `--extends` in CLI should override `extends` on `webpack.config.js` (I think eslint works the same, but need check)

**Comment by alexander-akait** [comment]
Fixed https://github.com/webpack/webpack-cli/pull/3738
